ndd_slurm.py

```diff
@@ -126,9 +149,14 @@ def get_srun_cmd(args):
     return cmd
 
 
-def init_process(cmd):
-    process = subprocess.Popen(cmd, stdin=subprocess.PIPE)
-    process.stdin.close()
+def init_process(cmd, read_fd=None, write_fd=None):
```

Comment:
Maybe we should assert that both `read_fd` and `write_fd` cannot be non-empty, because current function implementation does not correctly handle this combination.